felix.go

```diff
@@ -186,6 +186,19 @@ configRetry:
 			time.Sleep(1 * time.Second)
 			continue configRetry
 		}
+
+		// We now have some config flags that affect how we configure the syncer.
+		// After loading the config from the datastore, reconnect, possibly with new
+		// config.  We don't need to re-load the configuration _again_ because the
+		// calculation graph will spot if the config has changed since we were initialised.
+		datastoreConfig = configParams.DatastoreConfig()
```

Comment:
I'm not fully persuaded by the last sentence here.  ISTM that there is a window - between the `loadConfigFromDatastore` above and creating the second datastore client - where the config could change and Felix wouldn't get any update about it.  If that's right, the window can be closed by repeating the `loadConfigFromDatastore` and `UpdateFrom` lines here.

In any case I'll merge this PR, as it's a tiny window - but please do consider and respond to this comment.